x-pack/plugins/cloud_security_posture/server/lib/telemetry/collectors/installation_stats_collector.ts

```diff
@@ -13,15 +13,69 @@ import {
   SO_SEARCH_LIMIT,
 } from '@kbn/fleet-plugin/common';
 import { agentPolicyService } from '@kbn/fleet-plugin/server/services';
-import type { CloudSecurityInstallationStats } from './types';
+import type {
+  CloudbeatConfigKeyType,
+  CloudSecurityInstallationStats,
+  SetupAccessOption,
+} from './types';
 import type { CspServerPluginStart, CspServerPluginStartDeps } from '../../../types';
 import { CLOUD_SECURITY_POSTURE_PACKAGE_NAME } from '../../../../common/constants';
 
+interface CredentialMappings {
+  'gcp.credentials.file': 'credentials file';
+  'gcp.credentials.json': 'credentials json';
+  shared_credential_file: 'credentials file';
+  role_arn: 'role';
+}
+
 const getEnabledInputStreamVars = (packagePolicy: PackagePolicy) => {
   const enabledInput = packagePolicy.inputs.find((input) => input.enabled);
   return enabledInput?.streams[0].vars;
 };
 
+const getEnabledIsSetupAutomatic = (packagePolicy: PackagePolicy) => {
+  const cloudbeatConfig: Record<CloudbeatConfigKeyType, string> = {
+    'cloudbeat/cis_aws': 'cloud_formation_template_url',
+    'cloudbeat/vuln_mgmt_aws': 'cloud_formation_template_url',
+    'cloudbeat/cis_gcp': 'cloud_shell_url',
+    'cloudbeat/cis_azure': 'arm_template_url',
+  };
+
+  const enabledInput = packagePolicy.inputs.find((input) => input.enabled);
+
+  if (!enabledInput) return false;
+
+  const configKey = cloudbeatConfig[enabledInput.type as CloudbeatConfigKeyType];
+
+  return !!configKey && !!enabledInput.config?.[configKey]?.value;
```

Comment:
I'm not sure about this check. Do you check if the key exists? that is always true, no?

Reply:
I agree. I noticed there are some inconsistencies the config input stream variable names between different products and deployment options. I can open a tech debt ticket to address these inconsistencies. Mainly input streams var `credentials.type` needs to be populated regardless of deployment mode or product.

Reply:
That's a good point! I don't need to add this check `!!configKey`. I'll remove this in a follow up PR.  If manual option is chosen, then an automated template url will not exist